sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -341,69 +343,160 @@ void initializeHttpRequest(BlockStmt body, HttpRequestContext method) {
     }
 
     private void appendQueryParams(BlockStmt body, HttpRequestContext method) {
-        // Iterate through the query parameters and append them to the url string if they are not null
-        if (!method.getQueryParams().isEmpty()) {
-            // Declare newUrl once
-            Statement newUrlDeclaration = StaticJavaParser.parseStatement("String newUrl;");
-            newUrlDeclaration.setComment(new LineComment("\n Append non-null query parameters"));
-            body.addStatement(newUrlDeclaration);
-            body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
-            body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
-
-            method.getQueryParams().entrySet().forEach(entry -> {
-                String key = entry.getKey();
-                HttpRequestContext.QueryParameter value = entry.getValue();
-                boolean isValueTypeString = method.getParameters()
-                    .stream()
-                    .anyMatch(parameter -> parameter.getName().equals(value.getValue())
-                        && "String".equals(parameter.getShortTypeName()));
-                if (value.shouldEncode()) {
-                    if (isValueTypeString) {
-                        String encodedKey = "UriEscapers.QUERY_ESCAPER.escape(\"" + key + "\")";
-                        String encodedValue = "UriEscapers.QUERY_ESCAPER.escape(" + value.getValue() + ")";
+        if (method.getQueryParams().isEmpty()) {
+            return;
+        }
+
+        // Add imports and declare variables
+        body.tryAddImportToParentCompilationUnit(LinkedHashMap.class);
+        body.addStatement("LinkedHashMap<String, Object> queryParamMap = new LinkedHashMap<>();");
+
+        // Iterate through query parameters
+        method.getQueryParams().forEach((key, value) -> {
+            if (CoreUtils.isNullOrEmpty(key)) {
+                // Skip null or empty keys
+                return;
+            }
+
+            String valueExpression = value.getValue();
+            boolean isValueTypeString = method.getParameters()
+                .stream()
+                .anyMatch(parameter -> parameter.getName().equals(valueExpression)
+                    && "String".equals(parameter.getShortTypeName()));
+            boolean isValueTypeListOfString = method.getParameters()
+                .stream()
+                .anyMatch(parameter -> parameter.getName().equals(valueExpression)
+                    && "List<String>".equals(parameter.getShortTypeName()));
```

Comment:
Mind adding some comments here to clarify what's going on, I'm a bit confused and will state how I'm reading this right now.

We loop through all query parameters, checking if the interface method has any dynamic query parameters whose names match the value of the query parameter.

That seems a bit odd to me, as shouldn't we be checking that the static query parameter name matches a dynamic query parameter name? Why is `value.getValue()` used rather than `key`?